ksp_plugin/flight_plan.cpp

```diff
@@ -460,23 +457,23 @@ absl::Status FlightPlan::ComputeSegments(
 }
 
 void FlightPlan::AddLastSegment() {
-  segments_.emplace_back(segments_.back()->NewForkAtLast());
+  segments_.emplace_back(trajectory_.NewSegment());
   if (anomalous_segments_ > 0) {
     ++anomalous_segments_;
   }
 }
 
 void FlightPlan::ResetLastSegment() {
-  segments_.back()->ForgetAfter(segments_.back()->Fork()->time);
+  auto const last_segment = segments_.back();
+  trajectory_.ForgetAfter(std::next(last_segment->begin()));
```

Comment:
Since it has different semantics with respect to forgetting the given point, I think `ForgetAfter` should be renamed to something like `ForgetFrom`: the name `ForgetAfter` is misleading, I would have expected no `next`.
We can do that after migrating to DiscreteTraject0ry if that’s easier.

Reply:
I think some of the identifiers will need adjusting, but we'll do that at the end.